sdks/java/io/amazon-web-services/src/test/java/org/apache/beam/sdk/io/aws/dynamodb/DynamoDBIOTest.java

```diff
@@ -199,7 +209,8 @@ public void testRetries() throws Throwable {
                         writeRequest -> KV.of(tableName, writeRequest))
                 .withRetryConfiguration(
                     DynamoDBIO.RetryConfiguration.create(4, Duration.standardSeconds(10)))
-                .withAwsClientsProvider(AwsClientsProviderMock.of(amazonDynamoDBMock)));
+                .withAwsClientsProvider(AwsClientsProviderMock.of(amazonDynamoDBMock))
+                .withOverwriteByPKeys(overwriteByPKeys));
```

Comment:
Can we do this in an independent test to not change the original write test, e.g. `testDeduplicateWriteItems` and produce some duplicate items that test the correct removal of keys behaviour. I am assuming that for the batch case the direct runner will batch them together, but I will let a further note when I confirm this 100%.

Reply:
Spent some time working on this but couldn't come up with anything. Any tips how to test if the batches are correct? I can't think of anyway to test it in a pipeline, as the end result written to the db is the same (the issue is on api level). Should I do it outside a pipeline, call processElement manually and check the batch size?